src/queryResult/singletonStore.ts

```diff
@@ -20,30 +25,57 @@ class QueryResultSingletonStore {
         return QueryResultSingletonStore.instance;
     }
 
-    // Method to set a value in the store
-    public set(key: string, value: any): void {
-        this.store.set(key, value);
+    // Method to set a nested value in the store
+    public set(mainKey: string, subKey: string, value: any): void {
+        let nestedMap = this.store.get(mainKey);
+        if (!nestedMap) {
+            nestedMap = new Map<string, any>();
+            nestedMap.set(subKey, value);
+            this.store.set(mainKey, nestedMap);
+        } else {
+            nestedMap.set(subKey, value);
+            this.store.set(mainKey, nestedMap);
+        }
     }
 
-    // Method to get a value from the store
-    public get<T>(key: string): T | undefined {
-        return this.store.get(key);
+    // Method to get a nested value from the store
+    public get<T>(mainKey: string, subKey: string): T | undefined {
+        const nestedMap = this.store.get(mainKey);
+        return nestedMap?.get(subKey);
     }
 
-    // Method to check if a key exists
-    public has(key: string): boolean {
-        return this.store.has(key);
+    // Method to check if a nested key exists
+    public has(mainKey: string, subKey: string): boolean {
+        return this.store.get(mainKey)?.has(subKey) ?? false;
     }
 
-    // Method to delete a key-value pair
-    public delete(key: string): boolean {
-        return this.store.delete(key);
+    // Method to delete a nested key-value pair
+    public delete(mainKey: string, subKey: string): boolean {
```

Comment:
Does this class have any unit tests?  That's important to add, especially given that such a core part of the extension relies on it.

Reply:
added